spec/datadog/profiling/crashtracker_spec.rb

```diff
@@ -0,0 +1,201 @@
+          crashtracker.reset_after_fork
+
+          expect(`pgrep -f libdatadog-crashtracking-receiver`.lines.size).to be 2
+
+          crashtracker.stop
+
+          expect(`pgrep -f libdatadog-crashtracking-receiver`.lines.size).to be 1
+        end
+      end
+    end
+  end
+
+  describe '#stop' do
+    subject(:stop) { crashtracker.stop }
+
+    context 'when _native_stop_crashtracker raises an exception' do
+      it 'logs the exception' do
+        expect(described_class).to receive(:_native_stop) { raise 'Test failure' }
+        expect(Datadog.logger).to receive(:error).with(/Failed to stop crash tracking: Test failure/)
+
+        stop
+      end
+    end
+
+    it 'stops the crash tracker' do
+      crashtracker.start
+
+      stop
+
+      expect(`pgrep -f libdatadog-crashtracking-receiver`).to be_empty
+    end
+  end
+
+  context 'integration testing' do
+    shared_context 'HTTP server' do
+      let(:server) do
+        WEBrick::HTTPServer.new(
+          Port: 0,
+          Logger: log,
+          AccessLog: access_log,
+          StartCallback: -> { init_signal.push(1) }
+        )
+      end
+      let(:hostname) { '127.0.0.1' }
+      let(:log) { WEBrick::Log.new(StringIO.new, WEBrick::Log::WARN) }
+      let(:access_log_buffer) { StringIO.new }
+      let(:access_log) { [[access_log_buffer, WEBrick::AccessLog::COMBINED_LOG_FORMAT]] }
+      let(:server_proc) do
+        proc do |req, res|
+          messages << req.tap { req.body } # Read body, store message before socket closes.
+          res.body = '{}'
+        end
+      end
+      let(:init_signal) { Queue.new }
+
+      let(:messages) { [] }
+
+      before do
+        server.mount_proc('/', &server_proc)
+        @server_thread = Thread.new { server.start }
+        init_signal.pop
+      end
+
+      after do
+        unless RSpec.current_example.skipped?
+          # When the test is skipped, server has not been initialized and @server_thread would be nil; thus we only
+          # want to touch them when the test actually run, otherwise we would cause the server to start (incorrectly)
+          # and join to be called on a nil @server_thread
+          server.shutdown
+          @server_thread.join
+        end
+      end
+    end
+
+    include_context 'HTTP server'
+
+    let(:request) { messages.first }
+    let(:port) { server[:Port] }
+
+    let(:exporter_configuration) { [:agent, "http://#{hostname}:#{port}"] }
+
+    it 'reports crashes via http' do
+      fork_expectations = proc do |status:, stdout:, stderr:|
+        expect(Signal.signame(status.termsig)).to eq('SEGV').or eq('ABRT')
+        expect(stderr).to include('[BUG] Segmentation fault')
+      end
+
+      expect_in_fork(fork_expectations: fork_expectations) do
+        crashtracker.start
+
+        Process.kill('SEGV', Process.pid)
+      end
+
+      crash_report = JSON.parse(request.body, symbolize_names: true)[:payload].first
+
+      expect(crash_report[:stack_trace]).to_not be_empty
```

Comment:
Could also check that the correct signal type was reported

Reply:
Added in a2ae7305e2cadbb30f80bca5d8be673f6dda88c2